artificial-scrum-master/apps/artificial.scrum.master/Artificial.Scrum.Master.ScrumIntegration/Mappers/UserStories/UserStoriesWithTasksMapper.cs

```diff
@@ -0,0 +1,46 @@
+using Artificial.Scrum.Master.ScrumIntegration.Features.Shared.GetStoriesWithTasks;
+using Artificial.Scrum.Master.ScrumIntegration.Features.Shared.Models.Task;
+using Artificial.Scrum.Master.ScrumIntegration.Features.Shared.Models.UserStory;
+
+namespace Artificial.Scrum.Master.ScrumIntegration.Mappers.UserStories;
+
+internal interface IUserStoriesWithTasksMapper
+{
+    GetStoriesWithTasksResponse MapUserStoriesWithTasksResponse(
+        List<UserStory> userStoriesWithTasks,
+        List<StoryTask> tasks);
+}
+
+internal class UserStoriesWithTasksMapper : IUserStoriesWithTasksMapper
+{
+    public GetStoriesWithTasksResponse MapUserStoriesWithTasksResponse(
+        List<UserStory> userStoriesWithTasks,
+        List<StoryTask> tasks)
+    {
+        var mappedUserStories = userStoriesWithTasks.OrderBy(us => us.SprintOrder)
+            .Select(us =>
+            {
+                var taskNames = tasks
```

Comment:
Weź wyciągnij to poza linq, masz tutaj multiple enumeration po całej liscie.
Zrób sobie na górze

```suggestion
var taskNames = tasks 
                    .Where(t =>
                        !string.IsNullOrEmpty(t.Subject) &&
                        t.UserStoryId is not null)
                    .GroupBy(t => t.UserStoryId)
                    .ToDictionary(t => t.Key, t => t.ToList());
```